scripts/Util.py

```diff
@@ -1325,12 +1321,16 @@ def timedOut(self, timeout):
                     current.driver.setInterrupt(True)
                     raise
                 except Expect.TIMEOUT:
-                    if watchDog and watchDog.timedOut(timeout):
-                        print(
-                            "process {0} is hanging - {1}".format(
-                                process, time.strftime("%x %X")
-                            )
-                        )
+                    assert watchDog
+                    if watchDog.timedOut(timeout):
+                        print("process {0} is hanging - {1}".format(process, time.strftime("%x %X")))
+
+                        # If we're running in CI, dump the stack trace and exit
+                        if os.getenv("CI"):
```

Comment:
When we're in CI we collect dump the stack then exit. If it's not CI then we can do it manually.